src/algebra/quandle.lean

```diff
@@ -117,6 +125,57 @@ localized "infixr (name := shelf_hom) ` →◃ `:25 := shelf_hom" in quandles
 
 open_locale quandles
 
+namespace unital_shelf
+
+open shelf
+open unital_shelf
+
+variables {S : Type*} [unital_shelf S]
+
+/--
+A monoid is *graphic* if, for all `x` and `y`, the *graphic identity*
+`(x * y) * x = x * y` holds.  For a unital shelf, this graphic
+identity holds.
+-/
+lemma graphic_identity {x y : S} : (x ◃ y) ◃ x = x ◃ y :=
+begin
+  have h : (x ◃ y) ◃ x = (x ◃ y) ◃ (x ◃ 1) := by rw act_one,
+  rw [h, ←shelf.self_distrib, act_one],
+end
+
+lemma idempotent {x : S} : (x ◃ x) = x :=
+begin
+  have h : (x ◃ 1) ◃ (x ◃ 1) = (x ◃ x) := by rw act_one,
+  rw [←h, ←shelf.self_distrib, act_one, act_one],
+end
+
+lemma xxy_eq_xy {x y : S} : x ◃ (x ◃ y) = x ◃ y :=
+begin
+  have h : x ◃ (x ◃ y) = (x ◃ 1) ◃ (x ◃ y) := by rw act_one,
+  rw [h, ←shelf.self_distrib, one_act],
+end
+
+/--
+The associativity of a unital shelf comes for free.
+-/
+lemma assoc {x y z : S} : (x ◃ y) ◃ z = x ◃ y ◃ z :=
+begin
+  rw [self_distrib, self_distrib, graphic_identity, xxy_eq_xy],
+end
+
+/--
+A unital shelf is a *graphic monoid* and in particular, a monoid.
+-/
+@[priority 100]
+instance : monoid S :=
+{ one := 1,
+  one_mul := λ x, by exact one_act,
+  mul_one:= λ x, by exact act_one,
+  mul := act,
+  mul_assoc := λ x y z, by exact assoc }
```

Comment:
This instance feels like a bad idea to me. `monoid` says more than "this object has an associative multiplication etc", it says "the canonical spelling for multiplication of this type is `*`.

As an example of how this can go wrong, at the extreme case, I could write
```lean
instance : monoid nat := {mul := (+), one := (0), mul_assoc := add_assoc, ... }
```

Reply:
That's a fair complaint.  I've removed the instance in 8a1940335b34cfd81569bf2ceb0e3c1ae140dd8a